cli/unipipe/commands/git.ts

```diff
@@ -0,0 +1,99 @@
+import { Command } from "../deps.ts";
+import { Repository } from "../repository.ts";
+
+interface GitOpts {
+  name: string;
+  email: string;
+  message: string;
+}
+
+export function registerGitCmd(program: Command) {
+  program
+    .command("git <cmd> [repo]")
+    .description("Runs Git to simplify pull/push commands")
+    .action(async (opts: GitOpts, cmd: string, repo: string | undefined) => {
+
+      const repository = new Repository(repo ? repo : ".");
+      opts.name = opts.name || "Unipipe CLI";
+      opts.email = opts.email || "unipipe-cli@meshcloud.io";
+      opts.message = opts.message || "Commit changes";
+
+      switch (cmd) {
+        case "pull":
+          await commandPull(repository);
+          break;
+        case "push":
+          await commandPush(repository, opts);
+          break;
+        default:
+          console.log(`Git command '${cmd}' is not found`);
+      }
+    });
+}
+
+async function commandPull(repo: Repository) {
+  const pullFastForward = await gitPullFastForward(repo.path);
+
+  if (!pullFastForward) {
+    await gitPullRebase(repo.path);
+  }
+}
+
+async function commandPush(repo: Repository, opts: GitOpts) {
+  const add = await gitAdd(repo.path);
+  if (!add) return;
+
+  const hasChanges = await gitDiffIndex(repo.path);
+
+  if (!hasChanges) {
```

Comment:
f: This seems to be the wrong way around, right? Only if there are changes, we want to do a commit and a push. If there are no changes, we simply do nothing, right? But it seems to be implemented differently. Was there a specific reason to do it like this?

Reply:
Fixed